src/common/font.c

```diff
@@ -56,33 +60,30 @@ font_width(struct font *font, const char *string)
 	return rectangle.width;
 }
 
-void
-font_buffer_update(struct lab_data_buffer **buffer, int max_width,
-	const char *text, struct font *font, float *color, double scale)
-{
-	if (*buffer) {
-		wlr_buffer_drop(&(*buffer)->base);
-		*buffer = NULL;
-	}
-	font_buffer_create(buffer, max_width, text, font, color, scale);
-}
-
 void
 font_buffer_create(struct lab_data_buffer **buffer, int max_width,
-	const char *text, struct font *font, float *color, double scale)
+	const char *text, struct font *font, float *color, const char *arrow,
+	double scale)
 {
+	assert(max_width);
+
 	if (!text || !*text) {
 		return;
 	}
 
-	PangoRectangle rect = font_extents(font, text);
-	if (max_width && rect.width > max_width) {
-		rect.width = max_width;
+	PangoRectangle text_extents = font_extents(font, text);
+	PangoRectangle arrow_extents = font_extents(font, arrow);
+
+	if (arrow) {
+		text_extents.width = max_width - arrow_extents.width;
```

Comment:
This could turn negative if `max_width` is set to `1` or something small like that.
Currently this can't happen because the title doesn't use the arrow and the menu width is hard-coded.
It feels like a bug lingering around waiting to be triggered in the future with dynamic menu-width or similar though.

Reply:
fixed